tfx_bsl/beam/run_inference_arrow.py

```diff
@@ -261,32 +268,60 @@ def update_metrics_with_cache(self):
         self._model_byte_size.update(self.model_byte_size_cache)
         self.model_byte_size_cache = None
 
-    def update(self, elements: List[Union[tf.train.Example,
-                                          tf.train.SequenceExample]],
-               latency_micro_secs: int) -> None:
+    def update(
+      self, elements: List[Union[str, bytes]], latency_micro_secs: int) -> None:
       self._inference_batch_latency_micro_secs.update(latency_micro_secs)
       self._num_instances.inc(len(elements))
       self._inference_counter.inc(len(elements))
       self._inference_request_batch_size.update(len(elements))
       self._inference_request_batch_byte_size.update(
-          sum(element.ByteSize() for element in elements))
+          sum(len(element) for element in elements))
 
-  def __init__(self, inference_spec_type: model_spec_pb2.InferenceSpecType):
+  # TODO (Maxine): just one col for now, later, will do a list of str
+  def __init__(
+    self, inference_spec_type: model_spec_pb2.InferenceSpecType,
+    process_column: Optional[str] = None):
     super(_BaseDoFn, self).__init__()
     self._clock = None
+    self._process_column = process_column
     self._metrics_collector = self._MetricsCollector(inference_spec_type)
 
   def setup(self):
     self._clock = _ClockFactory.make_clock()
 
   def process(
-      self, elements: List[Union[tf.train.Example, tf.train.SequenceExample]]
+      self, elements: pa.RecordBatch
   ) -> Iterable[Any]:
     batch_start_time = self._clock.get_current_time_in_microseconds()
-    outputs = self.run_inference(elements)
-    result = self._post_process(elements, outputs)
+    # TODO (Maxine): take process as a parameter, should it be part of inference spec?
+    # extract record batch from here, assuming first column
+
+    # what would record batch look like? (flatten or not)
+    # vs np.asarray(elements.column(0))
+    if len(elements.columns) == 1: 
+      serialized_examples = elements.column(0).to_pylist()
+    else: 
+      if self._process_column is None:
+        raise ValueError('Must pass in a process column with multi-column RecordBatch')
+
+      serialized_examples = None
+      for column_name, column_array in zip(elements.schema.names, elements.columns):
+        column_type = column_array.type
+        if column_name == self._process_column:
```

Comment:
We can have a constant value for process_column and require it exist in RecordBatch. Then it does not need to be passed in.